File: mobile/src/main/java/com/alexstyl/specialdates/events/namedays/calendar/resource/NoSpecialNamedays.java
package com.alexstyl.specialdates.events.namedays.calendar.resource;

import com.alexstyl.specialdates.date.DayDate;
import com.alexstyl.specialdates.events.namedays.NameCelebrations;
import com.alexstyl.specialdates.events.namedays.NamesInADate;

import java.util.Collections;
import java.util.List;

public enum NoSpecialNamedays implements SpecialNamedays {
    INSTANCE;

    @Override
    public NamesInADate getNamedayOn(DayDate date) {
        return new NamesInADate(date);
    }

    @Override
    public List<String> getAllNames() {
        return Collections.emptyList();
    }

    @Override
    public NameCelebrations getNamedaysFor(String name, int year) {
        return new NameCelebrations(name);
    }
}

File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/calendar/resource/NoSpecialNamedays.kt
package com.alexstyl.specialdates.events.namedays.calendar.resource

import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.events.namedays.NameCelebrations
import com.alexstyl.specialdates.events.namedays.NamesInADate
import com.alexstyl.specialdates.events.namedays.NoCelebrations
import com.alexstyl.specialdates.events.namedays.NoNamesInADate

object NoSpecialNamedays : SpecialNamedays {

    override fun getNamedayOn(date: Date): NamesInADate {
        return NoNamesInADate(date)
    }

    override fun getNamedaysFor(name: String, year: Int): NameCelebrations {
        return NoCelebrations(name)
    }

    override val allNames: List<String>
        get() = emptyList()

}
